### PR TITLE
Enhancement of missing public API documentation

### PR DESCRIPTION
**Pull request by benedikt-roth**
Follow-up of #8832

### CODE REVIEW

**Comment by rossgardt** [comment]
Thank you for the review, @crisbeto . I made according changes to the docs.

File: src/lib/dialog/dialog-ref.ts [hunk 1]
```diff
@@ -47,6 +47,7 @@ export class MatDialogRef<T, R = any> {
     public _containerInstance: MatDialogContainer,
     // @breaking-change 8.0.0 `_location` parameter to be removed.
     _location?: Location,
+    /** Unique ID that is automatically being generated on instantiation. */
     readonly id: string = `mat-dialog-${uniqueId++}`) {
```
Comment (rossgardt) [inline src/lib/dialog/dialog-ref.ts]: This one stays as it is, right? Since it is readonly.

**Comment by rossgardt** [comment]
@crisbeto I also changed that other occurrence.
Shall the docs on the one `id` attribute that is marked as `readonly` also be changed?

File: src/lib/dialog/dialog-content-directives.ts [hunk 2]
```diff
@@ -79,6 +80,10 @@ export class MatDialogClose implements OnInit, OnChanges {
   },
 })
 export class MatDialogTitle implements OnInit {
+  /**
+   * Element ID of the dialog title HTML element. If no ID is specified,
+   * one is being generated automatically on instantiation.
```
Comment (jelbourn) [inline src/lib/dialog/dialog-content-directives.ts]: "one is being generated automatically on instantiation" ->
"one is generated automatically"

File: src/lib/dialog/dialog-ref.ts [hunk 3]
```diff
@@ -47,6 +47,7 @@ export class MatDialogRef<T, R = any> {
     public _containerInstance: MatDialogContainer,
     // @breaking-change 8.0.0 `_location` parameter to be removed.
     _location?: Location,
+    /** Unique ID that is automatically being generated on instantiation. */
```
Comment (jelbourn) [inline src/lib/dialog/dialog-ref.ts]: "Unique ID that is automatically being generated on instantiation." ->
"Unique ID for the dialog"

File: src/lib/expansion/expansion-panel-header.ts [hunk 4]
```diff
@@ -65,6 +65,7 @@ export class MatExpansionPanelHeader implements OnDestroy, FocusableOption {
   private _parentChangeSubscription = Subscription.EMPTY;
 
   constructor(
+      /** Reference to the containing Expansion Panel instance. */
```
Comment (jelbourn) [inline src/lib/expansion/expansion-panel-header.ts]: It should either be "expansion panel" (lower-case) or "MatExpansionPanel"

File: src/lib/form-field/error.ts [hunk 5]
```diff
@@ -22,5 +22,9 @@ let nextUniqueId = 0;
   }
 })
 export class MatError {
+  /**
+   * Element ID of the dialog title HTML element. If no ID is specified,
+   * one is being generated automatically on instantiation.
```
Comment (jelbourn) [inline src/lib/form-field/error.ts]: "one is being generated automatically on instantiation" ->
"one is generated automatically"

File: src/lib/snack-bar/simple-snack-bar.ts [hunk 6]
```diff
@@ -31,6 +31,7 @@ export class SimpleSnackBar {
   data: {message: string, action: string};
 
   constructor(
+    /** Reference to the Snack Bar object */
```
Comment (jelbourn) [inline src/lib/snack-bar/simple-snack-bar.ts]: Either "snack bar" (lower-case) or "MatSnackBar"

"object" -> "instance"

**Pull request closed without merging**